File: ruby/Get_Sender.rb
require 'net/http'
require 'uri'
require 'json'
require 'base64'

app_id = "API_KEY"
app_sec = "API_SECRET"
app_hash = Base64.strict_encode64("#{app_id}:#{app_sec}")

base_url = "https://api-sms.4jawaly.com/api/v1/"

# Example query, you can modify
query = {
  "page_size" => 10,
  "page" => 1,
  "status" => 1,
  "return_collection" => 1
}

url = URI.parse(base_url + "account/area/senders?" + URI.encode_www_form(query))

headers = {
  "Accept" => "application/json",
  "Content-Type" => "application/json",
  "Authorization" => "Basic #{app_hash}"
}

http = Net::HTTP.new(url.host, url.port)
http.use_ssl = true
request = Net::HTTP::Get.new(url.request_uri, headers)

response = http.request(request)
response_json = JSON.parse(response.body) rescue {}

puts "HTTP code: #{response.code}"
puts "API code: #{response_json['code']}"

if response.code == '200' && response_json['code'] == 200
  if response_json['items'] && !response_json['items'].empty?
    response_json['items'].each do |item|
      puts "#{item['sender_name']} (default: #{item['is_default'] == 1})"
    end
  else
    puts "No senders found!"
  end
elsif response_json['message']
  puts "Error: #{response_json['message']}"
else
  puts "Error code: #{response.code}"
end

File: python/Get_Sender.py
import requests
import json
import base64

app_id = "API_KEY"
app_sec = "API_secret"
app_hash = base64.b64encode(f"{app_id}:{app_sec}".encode()).decode()
base_url = "https://api-sms.4jawaly.com/api/v1/"
query = {
    "page_size": 10,
    "page": 1,
    "status": 1,
    "return_collection": 1
}

url = base_url + "account/area/senders?" + "&".join(f"{k}={v}" for k, v in query.items())

headers = {
    "Accept": "application/json",
    "Content-Type": "application/json",
    "Authorization": f"Basic {app_hash}"
}

response = requests.get(url, headers=headers)
response_json = response.json()

senders_list = []
if response.status_code == 200 and "items" in response_json and response_json["items"]:
    for item in response_json["items"]:
        sender_name = item.get("sender_name", "")
        is_default = bool(item.get("is_default", 0))
        senders_list.append({
            "sender_name": sender_name,
            "is_default": is_default
        })
    print(json.dumps(senders_list, ensure_ascii=False, indent=2))
else:
    print(json.dumps({"message": "No senders found"}, ensure_ascii=False, indent=2))
